File: teckwah_project/main/server/models/postal_code_model.py
# teckwah_project/main/server/models/postal_code_model.py
from sqlalchemy import Column, String, Integer, Enum, ForeignKey
from sqlalchemy.orm import relationship
from app.config.database import Base


class PostalCode(Base):
    __tablename__ = "postal_code"

    postal_code = Column(String(5), primary_key=True)
    city = Column(String(100), nullable=True)
    county = Column(String(100), nullable=True)
    district = Column(String(100), nullable=True)

    # Relationships
    postal_code_details = relationship(
        "PostalCodeDetail", back_populates="postal_code_info"
    )

    def __repr__(self):
        return f"<PostalCode(code={self.postal_code}, city={self.city})>"


class PostalCodeDetail(Base):
    __tablename__ = "postal_code_detail"

    postal_code = Column(
        String(5), ForeignKey("postal_code.postal_code"), primary_key=True
    )
    warehouse = Column(Enum("SEOUL", "BUSAN", "GWANGJU", "DAEJEON"), primary_key=True)
    distance = Column(Integer, nullable=False)
    duration_time = Column(Integer, nullable=False)

    # Relationships
    postal_code_info = relationship("PostalCode", back_populates="postal_code_details")

File: teckwah-TMS/server/models/PostalCodeModel.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/Database');

const PostalCode = sequelize.define(
  'postal_code',
  {
    postal_code: {
      type: DataTypes.STRING(5),
      primaryKey: true,
      allowNull: false,
      comment: '우편번호',
    },
    city: {
      type: DataTypes.STRING(100),
      allowNull: true,
      comment: '도시',
    },
    county: {
      type: DataTypes.STRING(100),
      allowNull: true,
      comment: '군/구',
    },
    district: {
      type: DataTypes.STRING(100),
      allowNull: true,
      comment: '동/읍/면',
    },
  },
  {
    tableName: 'postal_code',
    timestamps: false,
  }
);

// 우편번호 상세 모델 정의
const PostalCodeDetail = sequelize.define(
  'postal_code_detail',
  {
    postal_code: {
      type: DataTypes.STRING(5),
      primaryKey: true,
      allowNull: false,
      comment: '우편번호',
      references: {
        model: 'postal_code',
        key: 'postal_code',
      },
    },
    warehouse: {
      type: DataTypes.ENUM('SEOUL', 'BUSAN', 'GWANGJU', 'DAEJEON'),
      primaryKey: true,
      allowNull: false,
      comment: '창고',
    },
    distance: {
      type: DataTypes.INTEGER,
      allowNull: false,
      comment: '거리(미터)',
    },
    duration_time: {
      type: DataTypes.INTEGER,
      allowNull: false,
      comment: '소요시간(초)',
    },
  },
  {
    tableName: 'postal_code_detail',
    timestamps: false,
    indexes: [
      {
        name: 'idx_warehouse_postal',
        fields: ['warehouse', 'postal_code'],
      },
    ],
  }
);

// 관계 설정
PostalCode.hasMany(PostalCodeDetail, {
  foreignKey: 'postal_code',
  sourceKey: 'postal_code',
  as: 'details',
});

PostalCodeDetail.belongsTo(PostalCode, {
  foreignKey: 'postal_code',
  targetKey: 'postal_code',
  as: 'postalCode',
});

module.exports = {
  PostalCode,
  PostalCodeDetail,
};